src/memory_service/fam_memory_service_direct.cpp

```diff
@@ -355,6 +360,37 @@ void Fam_Memory_Service_Direct::copy(uint64_t srcRegionId, uint64_t srcOffset,
     MEMORY_SERVICE_DIRECT_PROFILE_END_OPS(mem_direct_copy);
 }
 
+void Fam_Memory_Service_Direct::backup(uint64_t srcRegionId, uint64_t srcOffset,
+                                       string outputFile, uint64_t size) {
+    ostringstream message;
+    std::string outputFilename = fam_backup_path + "/" + outputFile;
+    struct stat sb;
+    if (stat(outputFilename.c_str(), &sb) == 0) {
```

Comment:
Probably it is better to move this check also to allocator (Later we can plug-in other backup engines at one place)